test/profiler/test_profiler.py

```diff
@@ -354,34 +377,41 @@ def trace_handler(p):
             trace_called_num += 1
 
         use_cuda = torch.profiler.ProfilerActivity.CUDA in supported_activities()
-        # Create a temp file to save execution trace data.
+        # Create a temp file to save execution trace and kineto data.
         fp = tempfile.NamedTemporaryFile('w+t', suffix='.et.json', delete=False)
         fp.close()
-        expected_loop_events = 0
-        et = ExecutionTraceObserver()
-        et.register_callback(fp.name)
+        kt = tempfile.NamedTemporaryFile(mode="w+t", suffix=".kineto.json", delete=False)
+        kt.close()
+
+        et = ExecutionTraceObserver().register_callback(fp.name)
         with profile(
             activities=supported_activities(),
             schedule=torch.profiler.schedule(
                 skip_first=3,
                 wait=1,
                 warmup=1,
-                active=2),
+                active=2,
+                repeat=1),
             on_trace_ready=trace_handler,
+            execution_trace_observer=et,
```

Comment:
Based on my comment below, if we were to extract the observer out to an interface like `ITraceObserver` then this param can just take a list (or single) object satisfying that interface. 

This allows users to just expose any observer (or a list of observers) in the profiler context manager. ExecutionTraceObserver (and hence the Chakra schema) just seem to be a schema our group is interested and settled on, but imo it can vary with the use cases.